.github/workflows/docker_build_tpls.yml

```diff
@@ -0,0 +1,132 @@
+name: Docker TPL build
+
+# Manually triggered
+on:
+  workflow_dispatch:
+    # No input required
+
+jobs:
+  clang-10:
+    runs-on: ubuntu-latest
+    name: Builds a Clang 10 docker image and extracts generated hostconfigs
+    env:
+      TAG: clang-10
+      REPO: axomllnl/tpls
```

Comment:
I think this should be just `axom/tpls` - https://hub.docker.com/r/axom/tpls

Reply:
Nice that fixed it! gcc-8 now works. now to fix clang...

Reply:
its not liking hdf5 for some reason....

Reply:
Serac had a very similar issue: https://dev.azure.com/llnl-serac/serac/_build/results?buildId=2166&view=logs&j=2742912e-f547-59b4-f336-65c28301f30c&t=cd7bfce2-9b19-5c4f-5f3a-913e42776d4e&l=1341

```
HDF5-DIAG: Error detected in HDF5 (1.8.21) MPI-process 0:
  #000: H5E.c line 1653 in H5Eset_auto2(): library initialization failed
    major: Function entry/exit
    minor: Unable to initialize object
  #001: H5.c line 210 in H5_init_library(): unable to initialize dataset interface
    major: Function entry/exit
    minor: Unable to initialize object
  #002: H5Dint.c line 141 in H5D_init(): interface initialization failed
    major: Function entry/exit
    minor: Unable to initialize object
  #003: H5Dint.c line 183 in H5D__init_interface(): can't get default dataset creation property list
    major: Dataset
    minor: Inappropriate type
HDF5-DIAG: Error detected in HDF5 (1.8.21) MPI-process 0:
  #000: H5P.c line 297 in H5Pcreate(): not a property list class
    major: Invalid arguments to routine
    minor: Inappropriate type


MFEM abort: Fatal NetCDF error: NetCDF: HDF error
 ... in function: void mfem::Mesh::ReadCubit(const char *, int &, int &)
 ... in file: mesh/mesh_readers.cpp:2187

application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
```

Reply:
Looking at this spec:
```
axom@develop%clang@10.0.0+cpp14~cuda~debug~devtools+fortran+hdf5+lua~mfem+mpi+openmp~python+raja~scr+shared+umpire build_type=RelWithDebInfo cuda_arch=none arch=linux-ubuntu18.04-x86_64
  ^conduit@master%clang@10.0.0~adios~doc~doxygen+fortran+hdf5+hdf5_compat+mpi~python+shared~silo+test~zfp arch=linux-ubuntu18.04-x86_64
     ^hdf5@1.8.21%clang@10.0.0~cxx~debug~fortran~hl~java+mpi+pic~shared~szip~threadsafe api=none patches=b61e2f058964ad85be6ee5ecea10080bf79e73f83ff88d1fa4b602d00209da9c arch=linux-ubuntu18.04-x86_64
```
I wonder if changing HDF5 to be a shared library might help (s.t. they're all the same - changing them to all static might be interesting too)

Reply:
this seems like an issue with multiple versions of hdf5 (or multiple links of hdf5 at play).

I think a mix of shared and static could be the source of woes.

Reply:
Turns out conduit needed to be locked down and wasn't in the ubuntu-18 packages.yaml